src/proto/h1/role.rs

```diff
@@ -69,6 +69,14 @@ where
 
     let span = trace_span!("parse_headers");
     let _s = span.enter();
+
+    if let Some(h1_header_read_timeout) = ctx.h1_header_read_timeout {
+        if ctx.h1_header_read_timeout_fut.is_none() {
+            debug!("setting h1 header read timeout timer");
+            *ctx.h1_header_read_timeout_fut = Some(Box::pin(tokio::time::sleep(h1_header_read_timeout)));
```

Comment:
Is there a way we can either:

- Reuse the `Sleep` future, by calling `reset` with a new deadline.
- Reuse the `Box`, by setting a new one instead of clearing the `Box`.

Either of those would cut down on having to make an extra allocation per request, and we can just have 1 per connection.

Reply:
I ended up reusing the same `Sleep`